src/commands/admin/admin-subcommands/help.ts

```diff
@@ -0,0 +1,138 @@
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+import fs from "node:fs";
+import path from "node:path";
+import { SlashCommandSubcommandBuilder, EmbedBuilder, ChatInputCommandInteraction, Collection } from "discord.js";
+import { Subcommand } from "../../../util/command-template.js";
+import { defaultErrorHandler } from "../../../util/error-handler.js";
+import { MAGENTA } from "../../../util/colours.js";
+
+/*
+ * The help command for all admin commands
+ *
+ * This command module differs from all other commands because it needs to load them
+ * to access the help embed of each command
+ */
+
+// The default help window to show when no command is specified
+const defaultEmbed: EmbedBuilder = new EmbedBuilder()
+    .setTitle("Admin Help")
+    .setDescription("Here are my administrator commands! Use `/admin help <command>` to get help for a specific command")
+    .setColor(MAGENTA);
+
+
+// ----- LOAD HELP COMMAND EMBEDS -----
+
+const commands: Collection<string, EmbedBuilder> = new Collection();
+
+// The the contents of this folder
+const cmdFiles: string[] = fs.readdirSync(__dirname).filter(f => f.endsWith(".js"));
+
+// The string used to accumulate all commands
+let cmdstr: string = "";
+
+// Look through each JS file in each command in this folder
+for (const file of cmdFiles) {
+    // Skip this file
+    if (file === "help.js") {
+        cmdstr += "`/admin help`, ";
+        continue;
+    }
+
+    // Get the path of the current directory (./src) > "commands" > category > command
+    const cmdPath: string = path.join(__dirname, file);
+    const cmd: Subcommand = require(cmdPath).command;
+    
+    // If the imported file is a valid command, add it
+    if ("data" in cmd && "execute" in cmd && "error" in cmd && "help" in cmd) {
+        const name: string = cmd.data.name;
+        // The colour is set to magenta here
+        commands.set(name, cmd.help.setColor(MAGENTA));
+        cmdstr += `\`/admin ${name}\`, `;
+    }
+}
+// Remove the dangling comma
+// eslint-disable-next-line no-magic-numbers
+cmdstr = cmdstr.slice(0, cmdstr.length - 2);
+
+defaultEmbed.addFields({ name: "Commands", value: cmdstr });
+
+const name: string = "help";
+
+// Help command embed
+const help: EmbedBuilder = new EmbedBuilder()
+    .setTitle("Help")
+    .setDescription("The help command which displays useful command information!")
+    .addFields(
+        { name: "Format", value: `\`/admin ${name} [command]\`` },
+        { name: "[command]", value: "Optional parameter. The command that you want to know more about" }
+    )
+    .setColor(MAGENTA);
+
+// Add the help field of the help command
+commands.set("help", help);
+
+// The command choices for the help command
+// with the required fields in each choice
+const choices = Array.from(commands.keys()).map((cmd: string): { name: string, value: string } => {
+    return { name: cmd, value: cmd };
+});
+
+export const command: Subcommand = {
```

Comment:
This is the `/admin help` command, which is similar to `/help` but only for the admin commands